File: src/arraysAndStrings/java/palindromePermutation/frequencyApproach.java
package arraysAndStrings.java.palindromePermutation;

public class frequencyApproach {

	public boolean palindromePermutation(String s) {
		int[] count = new int[26];
		for(int i = 0; i < s.length(); i++){
			char c = s.charAt(i);
			if('a' < c && c < 'z') count[s.charAt(i) - 'a']++;
		}

		boolean oneLimit = false;
		for(int i: count) {
			if (i % 2 != 0) {
				if (oneLimit) return false;
				oneLimit = true;
			}
		}
		return true;
	}
}

File: src/arraysAndStrings/python/palindromePermutation/frequencyApproach.py
def palindromePermutation(s: str) -> bool:
    count = [0] * 26
    for i in s:
        if ord('a') <= ord(i) <= ord('z'):
            count[ord(i) - ord('a')] += 1

    one_limit = False
    for c in count:
        if c % 2 == 1:
            if one_limit:
                return False
            else:
                one_limit = True
    return True
